Sources/BraintreePayPal/BTPayPalApprovalURLParser.swift

```diff
@@ -36,8 +36,8 @@ struct BTPayPalApprovalURLParser {
         }
     }
     
-    init?(body: BTJSON) {
-        if let payPalAppRedirectURL = body["agreementSetup"]["paypalAppApprovalUrl"].asURL() {
+    init?(body: BTJSON, linkType: String?) {
+        if linkType == "universal", let payPalAppRedirectURL = body["agreementSetup"]["paypalAppApprovalUrl"].asURL() {
```

Comment:
Nit for another time - maybe another indicator we should move this to an enum

Reply:
Actually, I don't think the `BTPayPalApprovalURLParser` class should care if the merchant has opted into universal links or not. It's only determining which URL type it got back from the GW response. 

The flow logic for falling back to the web flow already is taken care of in `BTPayPalClient`.

Reply:
As of Thursday, we were still getting the `paypalAppApprovalUrl` from the Gateway regardless of the `BTPayPalClient`. This may have been because we were working with a mock and an update may have been deployed to just allow our client side handling. Worth digging into but I was still getting this URL with the `PPBeta` app uninstalled.